lib/streamlit/runtime/credentials.py

```diff
@@ -102,6 +105,52 @@ def email_prompt() -> str:
 }
 
 
+def save_email(email):
```

Comment:
Can you type-annotate this (`def save_email(email: str) -> None:`) and also add a docstring?

(Also, should it maybe be called `_save_email`, or is it a public API?)

Reply:
Also, this needs be part of a unit test (we should test that `Credentials.save` results in a call to `requests.post` with the expected data).

Reply:
Changed name and added tests